server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -228,8 +230,20 @@ protected final DocsStats docsStats(IndexReader indexReader) {
             } catch (IOException e) {
                 logger.trace(() -> "failed to get size for [" + info.info.name + "]", e);
             }
+            docsWithIgnoredFields = tryGetNumberOfDocumentsWithIgnoredFields(readerContext);
         }
-        return new DocsStats(numDocs, numDeletedDocs, sizeInBytes);
+        return new DocsStats(numDocs, numDeletedDocs, sizeInBytes, docsWithIgnoredFields);
+    }
+
+    private long tryGetNumberOfDocumentsWithIgnoredFields(final LeafReaderContext readerContext) {
+        try {
+            return readerContext.reader().getSumDocFreq(IgnoredFieldMapper.NAME);
+        } catch (IOException e) {
+            logger.trace(() -> "IO error while getting the number of documents with ignored fields", e);
+        } catch (UnsupportedOperationException e) {
```

Comment:
This happens for `source only` indices which do not include inverted index and doc values.